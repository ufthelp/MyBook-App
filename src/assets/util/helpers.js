var axios = require('axios');

let books = [];
function getUpdatedList (list){
        let len = list.length;
        for(let i = 0; i< len; i++){
            list[i].status = 'Add';
        }
        return list;
    }
module.exports = {
    getBooks: function(query){
        return axios.get(`https://www.googleapis.com/books/v1/volumes?q=${query}`)
            .then(function(response){
                return getUpdatedList(response.data.items);
            })
    },
    addBookInList : function(book){
        books.push(book);
    },
    getAddedBooks : function(){
        return books;
    },
        
};
