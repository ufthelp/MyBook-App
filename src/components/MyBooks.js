import React, { Component } from 'react';
/*MyBooks component-Display added books in 'MyBooks' panel on right view of app
**Child Components : N/A
*/
class MyBooks extends Component {

  render() {
    return (<div id="book-cta">
				<h2>
					{ this.props.title } 
				</h2>
				{ 
					this.props.addedBooks.map((book) =>
					<div className="book-info" key={book.id}>
						<div className="thumbnail">
							<img 
								src={!book.volumeInfo.imageLinks ? 'No Image' : book.volumeInfo.imageLinks.thumbnail}
								alt={!book.volumeInfo.imageLinks ? 'No Image' : book.volumeInfo.title}/>
						</div>
						<div className="book-details">
							<h4> 
								{book.volumeInfo.title}
							</h4>
							<h5> 
								{ book.volumeInfo.authors && book.volumeInfo.authors.length > 0 ? 'by ' + book.volumeInfo.authors[0] : 'Author Info Not Available'} 
							</h5>
							<p className="book-desc"> 
								{ !book.volumeInfo.description?'Description Not Available':book.volumeInfo.description }
							</p>
							<code className="book-page"> 
								{ !book.volumeInfo.pageCount? '0':book.volumeInfo.pageCount } pages 
							</code>
						</div>
					</div>
				)}
            </div>
        )
  }
}
export default MyBooks;





