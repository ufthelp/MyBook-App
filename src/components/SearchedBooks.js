import React, { Component } from 'react';

/*Searched books component,used to add books to My Books Panel on Right
**Child Components : N/A
*/
class SearchedBooks extends Component {
  constructor(props) {
    super(props);
    this.state ={
        addBook : 'Add'
    }

  }
/**
 * addBook - Change the book status to 'added' & calls the add function of parent component (SearchBox > App)
 * @param {*} book 
 */
  addBook(book){
    book.status = 'Added';
    this.props.add(book);
  }

  render() {
    return (
        <ul className="books-list">
            {
            this.props.books.map((book) =>
                <li key={ book.id }>
                    <h4>{ book.volumeInfo.title}</h4>
                    <h5>{ book.volumeInfo.authors && book.volumeInfo.authors.length > 0 ? 'by ' + book.volumeInfo.authors[0] : ''} </h5>
                    <h6  className= {book.status === 'Add' ? 'action-label' : ''} onClick={book.status === 'Add' ? () => this.addBook(book) : void(0)}>{book.status} </h6>
                </li>
            )}
        </ul>
    )             
            
}
}
export default SearchedBooks;



