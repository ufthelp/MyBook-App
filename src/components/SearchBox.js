import React, { Component } from 'react';
import SearchedBooks from './SearchedBooks'
import util from '../assets/util/helpers'

/*Search book component - Search the api to find the books based on the text supplied by user
**Child Components : SearchedBooks
*/
class SearchBox extends Component {
    constructor(props) {
        super();
        // This binding is necessary to make `this` work in the callback
        this.state = {
            books: null,
            query: 'React'
        };
        this.findBook = this.findBook.bind(this);
    }
    /*Search a book and calls the google api*/
    findBook(e){
        e.preventDefault();
        let query = this.refs.query.value;
        if(query !== ''){
            util.getBooks(query)
            .then(function(resp){
                this.setState({books : resp});
            }.bind(this));
        }
    }

  render() {
    return (<div id="left-nav">
        <input type="text" placeholder="Find a book" ref="query" onChange={this.findBook}/>
         { !this.state.books ? ''  : <SearchedBooks books={this.state.books} add={this.props.add}/> }
        </div>
        )
  }
}
export default SearchBox;

