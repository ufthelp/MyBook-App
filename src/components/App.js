import React, { Component } from 'react';
import '../assets/css/App.css';
import SearchBox from './SearchBox';
import MyBooks from './MyBooks';
import AppHeader from './AppHeader';
import util from '../assets/util/helpers'

/*Main APP component
**Child Components : SearchBox, MyBooks
*/
class App extends Component {
	constructor(props){
		super(props);
		this.onAdd = this.onAdd.bind(this);
		this.state = {
			addedBooks: [],
			title: 'No Books Added'
		};
	}

  /*Adding books to MyBooks list */
	onAdd(book){
		util.addBookInList(book);
		this.setState({addedBooks : util.getAddedBooks(),title: 'My Books'});
    }

  render() {
    return (
      <div className="App">
        <header>
          <AppHeader />
        </header>
        <main>
        	<SearchBox add={this.onAdd}/>
			<MyBooks addedBooks={this.state.addedBooks} title={this.state.title}/>	
        </main>
      </div>
    );
  }
}

export default App;
