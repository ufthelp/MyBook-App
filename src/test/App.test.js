import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import MyBooks from '../components/MyBooks';
import SearchBox from '../components/SearchBox';
import SearchedBooks from '../components/SearchedBooks';
import util from '../assets/util/helpers';
import Renderer from 'react-test-renderer'


/*App Level Test*/
describe('Test the App rendering and Snapshot',() =>{
	it('App renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	});

	test('Test App Snapshot',() =>{
		const component = Renderer.create(<App/>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();

	});
});

/*Test the rendering of child components*/
describe('Child level components are created',() =>{
  /*Test cases for MyBooks*/
  it('MyBooks Component is created', () => {
    const div = document.createElement('div');
    let addedBooks = [];
    ReactDOM.render(<MyBooks addedBooks ={addedBooks} title={"Yardi Book"} />, div);
  });
    /*Test cases for SearchedBooks*/
  it('SearchedBooks Component is rendered', () => {
    const div = document.createElement('div');
    let books =[];
    ReactDOM.render(<SearchedBooks books={books}/>, div);
  });
  /*Test cases for SearcheBox*/
  it('SearchBox Component is rendered', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBox/>, div);
  });
});

