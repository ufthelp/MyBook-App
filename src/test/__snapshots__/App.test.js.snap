// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App rendering and Snapshot Test App Snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to My Book App
    </h2>
  </div>
  <main>
    <div
      id="left-nav"
    >
      <input
        onChange={[Function]}
        placeholder="Find a book"
        type="text"
      />
      
    </div>
    <div
      id="book-cta"
    >
      <h2>
        No Books Added
          
      </h2>
    </div>
  </main>
</div>
`;
